src/components/Swap/Settings/index.tsx

```diff
@@ -1,72 +1,44 @@
-import { Trans } from '@lingui/macro'
 import { Settings as SettingsIcon } from 'icons'
-import { useAtomValue, useResetAtom } from 'jotai/utils'
-import { useCallback, useState } from 'react'
-import { swapEventHandlersAtom } from 'state/swap'
-import { settingsAtom } from 'state/swap/settings'
+import { useState } from 'react'
 import styled from 'styled-components/macro'
-import { ThemedText } from 'theme'
 
-import { IconButton, TextButton } from '../../Button'
+import { IconButton } from '../../Button'
 import Column from '../../Column'
-import Dialog, { Header } from '../../Dialog'
-import { BoundaryProvider } from '../../Popover'
+import Popover, { BoundaryProvider } from '../../Popover'
 import MaxSlippageSelect from './MaxSlippageSelect'
 import TransactionTtlInput from './TransactionTtlInput'
 
-export function SettingsDialog() {
+export function SettingsPopover() {
   const [boundary, setBoundary] = useState<HTMLDivElement | null>(null)
-  const { onSettingsReset } = useAtomValue(swapEventHandlersAtom)
-  const resetSettingsBase = useResetAtom(settingsAtom)
-  const resetSettings = useCallback(() => {
-    onSettingsReset?.()
-    resetSettingsBase()
-  }, [onSettingsReset, resetSettingsBase])
+
   return (
-    <>
-      <Header title={<Trans>Settings</Trans>} ruled>
-        <TextButton onClick={resetSettings}>
-          <ThemedText.ButtonSmall>
-            <Trans>Reset</Trans>
-          </ThemedText.ButtonSmall>
-        </TextButton>
-      </Header>
-      <Column gap={1} style={{ paddingTop: '1em' }} ref={setBoundary} padded>
-        <BoundaryProvider value={boundary}>
-          <MaxSlippageSelect />
-          <TransactionTtlInput />
-        </BoundaryProvider>
-      </Column>
-    </>
+    <Column gap={1} style={{ paddingTop: '1em' }} ref={setBoundary} padded>
+      <BoundaryProvider value={boundary}>
+        <MaxSlippageSelect />
+        <TransactionTtlInput />
+      </BoundaryProvider>
+    </Column>
   )
 }
 
-const SettingsButton = styled(IconButton)<{ hover: boolean }>`
-  ${SettingsIcon} {
-    transform: ${({ hover }) => hover && 'rotate(45deg)'};
-    transition: ${({ hover }) => hover && 'transform 0.25s'};
-    will-change: transform;
+const SettingsButton = styled(IconButton)`
+  ${SettingsIcon}:hover {
+    transform: rotate(45deg);
+    transition: transform 0.25s;
   }
 `
 
-export default function Settings({ disabled }: { disabled?: boolean }) {
+export default function Settings() {
   const [open, setOpen] = useState(false)
-  const [hover, setHover] = useState(false)
+  const [wrapper, setWrapper] = useState<HTMLDivElement | null>(null)
```

Comment:
for my sake, can you explain the benefit of using `useState` instead of `useRef` here? i have never seen it before and want to understand.

Reply:
oo this is copied from other places where we have `BoundaryProvider`s, i think in general they behave similarly except in the `useState` version, the `wrapper` value will actually cause a rerender to `BoundaryProvider` since updates to `wrapper` are not referentially stable. havent dug too much into how these boundary providers work yet tho~

Reply:
It is so that the component will rerender once the wrapper is set ([see docs](https://reactjs.org/docs/hooks-reference.html#:~:text=Keep%20in%20mind%20that%20useRef%20doesn%E2%80%99t%20notify%20you%20when%20its%20content%20changes.%20Mutating%20the%20.current%20property%20doesn%E2%80%99t%20cause%20a%20re%2Drender.%20If%20you%20want%20to%20run%20some%20code%20when%20React%20attaches%20or%20detaches%20a%20ref%20to%20a%20DOM%20node%2C%20you%20may%20want%20to%20use%20a%20callback%20ref%20instead.)).